### ISSUE TITLE
Promise Error in code but builds and runs fine

### ISSUE THREAD

**Issue by chelvaric** [angular/angular#36787]
# 🐞 bug report

### Affected Package
@angular/router

### Is this a regression?
didn't have the error in angular 8

### Description

My app runs fine so its not a huge problem but wherever i wan't to use promise it complains in the code with :
 A dynamic import call returns a 'Promise'. Make sure you have a declaration for 'Promise' or include 'ES2015' in your `--lib` option.ts(2711)

like example in import of the router module it complains but everything builds and works

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/issue-promise-angular9

## 🔥 Exception or Error
<pre><code>
 loadChildren: () => import('./home/home.module').then( m => m.HomePageModule),
A dynamic import call returns a 'Promise'. Make sure you have a declaration for 'Promise' or include 'ES2015' in your `--lib` option.ts(2711
</code></pre>

## 🌍  Your Environment

**Angular Version:** 9
<pre><code>
Angular CLI: 9.1.3
Node: 12.16.1
OS: win32 x64

Angular: 9.1.3
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.3
@angular-devkit/build-angular     0.901.3
@angular-devkit/build-optimizer   0.901.3
@angular-devkit/build-webpack     0.901.3
@angular-devkit/core              8.3.26
@angular-devkit/schematics        8.3.26
@ngtools/webpack                  9.1.3
@schematics/angular               8.3.26
@schematics/update                0.901.3
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

</code></pre>

its all in vscode

**Comment by atscott** [comment]
Hi @chelvaric, I don't see the error/warning you're describing in the stackblitz example. Could you please provide minimal reproduction as a GitHub repo with steps to reproduce?

**Comment by atscott** [comment]
Looking at your `package.json` in the stackblitz example, it appears you need "es2015.promise" in the "lib" list. See related answer on stackoverlow: https://stackoverflow.com/a/38212418

**Comment by chelvaric** [comment]
yes in stackblitz it is fine, i had it in visual studio code, But why do we need to add the es2015.promise ourselfes in a clean project? shoudln't it be added standard then?

**Comment by atscott** [comment]
@chelvaric It's hard to know exactly what's happening for you without specific steps to reproduce the issue. I tried downloading the project from stackblitz, running `npm install`, and opening it in vs code but did not see the warning you're describing. I also created a new project with the CLI using these steps:
1. `ng new test` and add routing when prompted
2. `cd test`
3. `ng g m home`
4. add `{path: '', loadChildren: () => import('./home/home.module').then( m => m.HomeModule)}` to `app-routing.module.ts`
I did not see the warning in this scenario either so this doesn't appear to be an issue with Angular or the CLI but something specific about your project/workspace. It may be that your npm packages need to be re-installed.
